src/d3a/models/area/__init__.py

```diff
@@ -127,27 +128,39 @@ def area_reconfigure_event(self, **kwargs):
                 if key_in_dict_and_not_none(kwargs, 'grid_fee_constant') else 0
             grid_fee_percentage = kwargs["grid_fee_percentage"] \
                 if key_in_dict_and_not_none(kwargs, 'grid_fee_percentage') else 0
-
-            validate_area(grid_fee_percentage=grid_fee_percentage,
-                          grid_fee_constant=grid_fee_constant)
-            self._set_grid_fees(grid_fee_constant, grid_fee_percentage)
+            try:
+                validate_area(grid_fee_percentage=grid_fee_percentage,
+                              grid_fee_constant=grid_fee_constant)
+                self._set_grid_fees(grid_fee_constant, grid_fee_percentage)
+            except D3AException as e:
+                log.error(str(e))
 
         if key_in_dict_and_not_none(kwargs, 'baseline_peak_energy_import_kWh'):
-            self.baseline_peak_energy_import_kWh = kwargs['baseline_peak_energy_import_kWh']
-            validate_area(baseline_peak_energy_import_kWh=self.baseline_peak_energy_import_kWh)
+            try:
+                validate_area(baseline_peak_energy_import_kWh=self.baseline_peak_energy_import_kWh)
+                self.baseline_peak_energy_import_kWh = kwargs['baseline_peak_energy_import_kWh']
+            except D3AException as e:
+                log.error(str(e))
+
         if key_in_dict_and_not_none(kwargs, 'baseline_peak_energy_export_kWh'):
-            self.baseline_peak_energy_export_kWh = kwargs['baseline_peak_energy_export_kWh']
-            validate_area(baseline_peak_energy_export_kWh=self.baseline_peak_energy_export_kWh)
+            try:
+                validate_area(baseline_peak_energy_export_kWh=self.baseline_peak_energy_export_kWh)
+                self.baseline_peak_energy_export_kWh = kwargs['baseline_peak_energy_export_kWh']
+            except D3AException as e:
+                log.error(str(e))
 
         if key_in_dict_and_not_none(kwargs, 'import_capacity_kVA') or \
                 key_in_dict_and_not_none(kwargs, 'export_capacity_kVA'):
             import_capacity_kVA = kwargs["import_capacity_kVA"] \
                 if key_in_dict_and_not_none(kwargs, 'import_capacity_kVA') else None
             export_capacity_kVA = kwargs["export_capacity_kVA"] \
                 if key_in_dict_and_not_none(kwargs, 'export_capacity_kVA') else None
-            validate_area(import_capacity_kVA=import_capacity_kVA,
-                          export_capacity_kVA=export_capacity_kVA)
-            self._convert_area_throughput_kva_to_kwh(import_capacity_kVA, export_capacity_kVA)
+            try:
+                validate_area(import_capacity_kVA=import_capacity_kVA,
```

Comment:
Optional, but IIRC we discussed about being able to not apply the whole event if the validation fails, this is not followed here since the event is only partially applied if, e.g. the validation fails the `baseline_peak_energy_export_kWh` check. Had we settled on requirements about that ?

Reply:
So you mean to follow the same approach as in the device `area_reconfigure_event`: Do validate all the inputs first and only if there are no exceptions, we update? 
I cant remember discussing this, but i totally agree and will change it.

Reply:
Exactly! Thanks!